src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php

```diff
@@ -25,7 +25,7 @@ private function filterFilesByExtension(Traversable $files, string $fileExtensio
 
         /* @var $file \SplFileInfo */
         foreach ($files as $file) {
-            if (! preg_match($extensionMatcher, $file->getBasename())) {
+            if ($file->isDir() || ! preg_match($extensionMatcher, $file->getBasename())) {
```

Comment:
I've added it a performance tweak, I take it out as it is not needed for this PR :+1: